usr/lib/pkcs11/ica_s390_stdll/ica_specific.c

```diff
@@ -120,33 +120,76 @@ ica_ec_key_free_t               p_ica_ec_key_free;
 static int ica_ec_support_available = 0;
 
 static CK_RV ecc_support_in_libica_available(void)
+{
+    if (p_ica_ec_key_new != NULL &&
+        p_ica_ec_key_init  != NULL &&
+        p_ica_ec_key_generate  != NULL &&
+        p_ica_ecdh_derive_secret  != NULL &&
+        p_ica_ecdsa_sign  != NULL &&
+        p_ica_ecdsa_verify  != NULL &&
+        p_ica_ec_key_get_public_key  != NULL &&
+        p_ica_ec_key_get_private_key  != NULL &&
+        p_ica_ec_key_free != NULL)
+        return 1;
+
+    return 0;
+}
+#endif
+
+#ifdef SHA512_224
```

Comment:
where is SHA512_224 etc defined? in ica_api.h ?

Reply:
Yes in ica_api.h . That way I detect an older version libica and build the ICA token without support for SHA-512/224/256.